crates/sp1/evm-exec/script/src/bin/datagen.rs

```diff
@@ -52,9 +53,11 @@ struct Args {
 /// Loads the genesis file from disk and converts it into a ChainSpec
 fn load_chain_spec_from_genesis(path: &str) -> Result<(Genesis, Arc<ChainSpec>), Box<dyn Error>> {
     let genesis_json = fs::read_to_string(path).wrap_err_with(|| format!("Failed to read genesis file at {}", path))?;
+    let alloy_genesis: AlloyGenesis = serde_json::from_str(&genesis_json)?;
+
+    let genesis = Genesis::Custom(alloy_genesis.config);
+    let chain_spec: Arc<ChainSpec> = Arc::new((&genesis).try_into()?);
```

Comment:
Looks simple but this was the culprit which was a nightmare to debug. 

The genesis deserialisation was silently failing and falling back to defaults (e.g. chain id 1, mainnet) which was causing errors way further down the stack. 
[Here to be specific](https://github.com/paradigmxyz/reth/blob/main/crates/ethereum/consensus/src/lib.rs#L181-L185).